ansible_sdk/_util/metrics.py

```diff
@@ -84,13 +86,17 @@ async def collect_metrics(self, status_obj: AnsibleJobStatus):
 
         job_headers = [x.name for x in dataclasses.fields(AnsibleJobStats)]
         module_headers = [x.name for x in dataclasses.fields(AnsibleModuleStats)]
-        collection_headers = [x.name for x in dataclasses.fields(AnsibleCollectionStats)]
+        collection_headers = [
+            x.name for x in dataclasses.fields(AnsibleCollectionStats)
+        ]
         role_headers = [x.name for x in dataclasses.fields(AnsibleRoleStats)]
 
-        with open(job_csv_filename, "w") as job_csv_fh, \
-                open(collection_csv_filename, "w") as collection_csv_fh, \
-                open(roles_csv_filename, "w") as role_csv_fh, \
-                open(module_csv_filename, "w") as module_csv_fh:
+        with (
+            open(job_csv_filename, "w") as job_csv_fh,
```

Comment:
Forgot before: these `open`s are all also blocking- I'd suggest just using a threadpool shim over `open` rather than using `aiofiles`, since the underlying File I/O from the CSVWriter will still be using classic blocking stuff.